### ISSUE TITLE
cannot run make all to generate redisgear.so

### ISSUE THREAD

**Issue by Tigojiang** [RedisGears/RedisGears#576]
Hi, the error is shown as followed. When running make all to generate the redisgear.so, it logs Symbol not found: _libiconv. The Mac is of 10.15.6.
<img width="871" alt="截屏2021-07-05 下午5 11 07" src="https://user-images.githubusercontent.com/54599335/124447479-fdb03480-ddb3-11eb-8773-45817cd49c63.png">

**Comment by MeirShpilraien** [comment]
@rafie @chayim any input on this?

**Comment by chayim** [comment]
Hi @Tigojiang can I have your build details? Testing on Catalina **10.15.7**, the libraries are found below, having first run a **make all**.  Can you please provide the outputs of _otool -L <each library>.so_ for both the gears_python.so and redisgears.so?

Final question: Did you install the dependencies via our system-setup.py script, or some other method?  For reference, below are the outputs from my test system.

```
otool -L gears_python.so
gears_python.so:
        ../../bin/macos-x64-release/gears_python/gears_python.so (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
ec2-user@ip-172-31-20-187 gears_python % cd ../
ec2-user@ip-172-31-20-187 macos-x64-release % ls
cpython          gears_python     hiredis          libevent         python3_99.99.99 redisgears.so    snapshot         src
ec2-user@ip-172-31-20-187 macos-x64-release % otool -L redisgears.so 
redisgears.so:
        bin/macos-x64-release/redisgears.so (compatibility version 0.0.0, current version 0.0.0)
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1677.104.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
```

**Comment by Tigojiang** [comment]
Hello, this is the output. I use the gear's system-setup.py. Because I follow the steps in the quickstart guide. Make setup, make fetch, make, all. @chayim 
<img width="790" alt="截屏2021-07-06 上午9 30 21" src="https://user-images.githubusercontent.com/54599335/124529284-cb96e500-de3c-11eb-9b84-b6a9b366df11.png">

**Comment by Tigojiang** [comment]
Hi, does this output look correct? Probably they are the same. @chayim

**Comment by chayim** [comment]
They're identical, and I can't reproduce. The sole exception is that my environment is 10.15.7 and yours is 10.15.6. Can you attach your binaries here? I'm happy to have a deeper dive.

**Comment by Tigojiang** [comment]
Hello, do you mean the binary files of redisgear.so. Sorry I am not quite familiar with these terms now lol. Thank you very much! @chayim

**Comment by chayim** [comment]
Yes please @Tigojiang. Mind attaching to this a zip file containing the shared objects (.so files) generated by your build? Specifically I'd like gears\_python.so and redisgears.so if you don't mind.

**Comment by Tigojiang** [comment]
Hi! This is the file! @chayim 
[files with .so.zip](https://github.com/RedisGears/RedisGears/files/6798055/files.with.so.zip)

**Comment by MeirShpilraien** [comment]
@chayim any news here?
